handler/population_types.go

```diff
@@ -163,6 +164,57 @@ func (h *PopulationTypes) Get(w http.ResponseWriter, req *http.Request) {
 	h.respond.JSON(ctx, w, http.StatusOK, resp)
 }
 
+func (h *PopulationTypes) Get(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	ptype := chi.URLParam(r, "population-type")
+	logData := log.Data{
+		"population_type": ptype,
+	}
+
+	ptypes, err := h.cantabular.ListDatasets(ctx)
+	if err != nil {
+		h.respond.Error(ctx, w, dperrors.StatusCode(err), &Error{
+			err:     errors.Wrap(err, "failed to fetch population types"),
+			message: "failed to get population types",
+		})
+		return
+	}
+
+	var resp contract.GetPopulationTypeResponse
+
+	for _, p := range ptypes.Datasets {
+		if p.Name == ptype {
+			resp.PopulationType = contract.PopulationType{
+				Name:        p.Name,
+				Label:       p.Label,
+				Description: p.Description,
+			}
+			break
+		}
+	}
+
+	if resp.PopulationType.Name == "" {
+		h.respond.Error(ctx, w, http.StatusNotFound, &Error{
+			err:     errors.New("population type not found"),
+			logData: logData,
+		})
+		return
+	}
+
+	if !h.cfg.EnablePrivateEndpoints {
```

Comment:
It will be preferable to move this check to line 173 just after ptype is extracted from url param